docs/Training-On-Your-Data.md

```diff
@@ -0,0 +1,354 @@
+`model_axis_size`. (For most models `model_axis_size` should be 1, so you can ignore it.)
+
+`per_device_parallelism` is analogous to `per_device_batch_size` in other frameworks. It controls how many examples
+are processed at once on a single accelerator, but the name is a bit more "correct" in the presence of tensor or pipeline
+parallelism.  A `per_device_parallelism` of `-1` means "use as many as possible to not perform gradient accumulation."
+
+Gradient accumulation is performed whenever `num_accelerators * per_device_parallelism / model_axis_size < train_batch_size`.
+Gradient checkpointing is enabled by default and highly recommended.
+
+So, to find your batch size, you should modify either `per_device_parallelism` or `train_batch_size` until
+your job runs. Note that, due to FSDP, as you add more TPUs, you can increase the effective parallelism because
+you will use less memory per accelerator to store parameters and optimizer states.
+
+### Number of Training Steps
+
+Levanter does not support epochs or number of tokens/examples, so if you want to train for a certain number of epochs or
+tokens, you'll need to compute the number of steps yourself. You can use the following formula:
+
+```
+num_train_steps = num_epochs * num_tokens_per_epoch / train_batch_size / seq_len
+```
+
+Note however that epoch boundaries aren't really respected: our implementation of sharded data loading restarts
+from the beginning as soon as any machine finishes its shards.
+
+## Launching Training
+
+### TPU
+
+First, we assume you've gone through the setup steps in [the TPU guide](./Getting-Started-TPU-VM.md), at least through setting up your gcloud account.
+We also strongly recommend setting up ssh keys and ssh-agent.
+
+#### Upload Config To GCS
+
+Once you have your config built, you should upload it to GCS. You could also `scp` it to all workers, but this is easier
+and works with the TPU babysitting script.
+
+```bash
+gsutil cp my_config.yaml gs://path/to/config.yaml
+```
+
+#### Using the Babysitting Script with a Preemptible or TRC TPU VM
+
+If you are using a preemptible TPU VM, or a TRC TPU VM, you should use the babysitting script to automatically restart
+your VM if it gets preempted. A detailed guide to babysitting is available in the
+[babysitting section of the TPU guide](./Getting-Started-TPU-VM.md#using-the-babysitting-script-with-a-preemptible-or-trc-tpu-vm).
+Here is the upshot:
+
+```bash
+infra/babysit-tpu-vm my-tpu -z us-east1-d -t v3-128 -- \
+    WANDB_API_KEY=... levanter/infra/run.sh python levanter/src/levanter/main/train_lm.py --config_path gs://path/to/config.yaml
+```
+
+#### Spin up and manual launch
+
+You should probably use the automated setup script, as described in the [relevant section of the TPU guide](./Getting-Started-TPU-VM.md#automatic-setup).
+Here's what that looks like:
+
+```bash
+bash infra/spin-up-tpu-vm.sh my-tpu -z us-east1-d -t v3-128
```

Comment:
```suggestion
bash infra/spin-up-vm.sh my-tpu -z us-east1-d -t v3-128
```

Also, does this run setup? I thought we will need to attach the set up script, like:

```
bash infra/spin-up-vm.sh $NAME -b $BRANCH -z $ZONE -t $TYPE -s infra/setup-tpu-vm.sh
```